dotnet/samples/Concepts/Memory/HuggingFace_TextEmbeddingCustomHttpHandler.cs

```diff
@@ -0,0 +1,75 @@
+/// Generally, an embedding model will return results as a 1 * n matrix for input type [string]. However, the model can have different matrix dimensionality.
+/// For example, the <a href="https://huggingface.co/cointegrated/LaBSE-en-ru">cointegrated/LaBSE-en-ru</a> model returns results as a 1 * 1 * 4 * 768 matrix, which is different from Hugging Face embedding generation service implementation.
+/// To address this, a custom <see cref="HttpClientHandler"/> can be used to modify the response before sending it back.
+/// </summary>
+public class HuggingFace_TextEmbeddingCustomHttpHandler(ITestOutputHelper output) : BaseTest(output)
+{
+    public async Task RunInferenceApiEmbeddingCustomHttpHandlerAsync()
+    {
+        Console.WriteLine("\n======= Hugging Face Inference API - Embedding Example ========\n");
+
+        var hf = new HuggingFaceTextEmbeddingGenerationService(
+            "cointegrated/LaBSE-en-ru",
+            apiKey: TestConfiguration.HuggingFace.ApiKey,
+            httpClient: new HttpClient(new CustomHttpClientHandler()
+            {
+                CheckCertificateRevocationList = true
+            })
+        );
+
+        var sqliteMemory = await SqliteMemoryStore.ConnectAsync("./../../../Sqlite.sqlite");
+
+        var skMemory = new MemoryBuilder()
+            .WithTextEmbeddingGeneration(hf)
+            .WithMemoryStore(sqliteMemory)
+            .Build();
+
+        if (!await sqliteMemory.DoesCollectionExistAsync("Sqlite"))
+        {
+            await sqliteMemory.CreateCollectionAsync("Sqlite");
+        }
+
+        await skMemory.SaveInformationAsync("Test", "THIS IS A SAMPLE", "sample", "TEXT");
+    }
+
+    private sealed class CustomHttpClientHandler : HttpClientHandler
+    {
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            // Log the request URI
+            //Console.WriteLine($"Request: {request.Method} {request.RequestUri}");
+
+            // Send the request and get the response
+            HttpResponseMessage response = await base.SendAsync(request, cancellationToken);
+
+            // Log the response status code
+            //Console.WriteLine($"Response: {(int)response.StatusCode} {response.ReasonPhrase}");
+
+            // You can manipulate the response here
+            // For example, add a custom header
+            // response.Headers.Add("X-Custom-Header", "CustomValue");
+
+            // For example, modify the response content
+            string originalContent = await response.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(false);
+            var modifiedContent = JsonSerializer.Deserialize<List<List<List<ReadOnlyMemory<float>>>>>(originalContent);
+            var modifiedResponse = JsonSerializer.Serialize(modifiedContent[0][0].ToList());
+            response.Content = new StringContent(modifiedResponse);
```

Comment:
Should we use streams in this example to improve it's efficiency?  It will complicate it a little bit.
Something like this:
```cs
            Stream originalStream = await response.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
            var modifiedContent = await JsonSerializer.DeserializeAsync<List<List<List<ReadOnlyMemory<float>>>>>(originalStream, cancellationToken: cancellationToken).ConfigureAwait(false);

            Stream modifiedStream = new MemoryStream();
            await JsonSerializer.SerializeAsync(modifiedStream, modifiedContent[0][0].ToList(), cancellationToken: cancellationToken).ConfigureAwait(false);
            response.Content = new StreamContent(modifiedStream);
```

Reply:
Yes I also thought about that .
Convert stream to list then again convert back to stream this will lead to performance.

Reply:
Make the serialisation de-serialisation asynchronous